R/add_highlight.R

```diff
@@ -0,0 +1,136 @@
+#' @title Highlight a specific strata
+#'
+#' @description S3 method for highlighting a specific strata by lowering the opacity of all other strata.
+#'
+#' @author Tim Treis
+#'
+#' @param gg visR object
+#' @param ... other arguments passed on to the method 
+#' 
+#' @examples
+#' library(visR)
+#'
+#' adtte %>%
+#'   visR::estimate_KM("SEX") %>%
+#'   visR::visr() %>%
+#'   visR::add_CI(alpha = 0.4) %>%
+#'   visR::add_highlight("SEX=M", bg_alpha_multiplier = 0.2)
+#'
+#' @return The input `ggplot2` object with adjusted `alpha` values
+#'
+#' @rdname add_highlight
+#'
+#' @export
+
+add_highlight <- function(gg, ...){
+  UseMethod("add_highlight", gg)
+}
+
+#' @param gg A ggplot created with visR
+#' @param strata Name of the strata to be highlighted as shown in the legend.
+#' @param bg_alpha_multiplier Factor with which the `alpha` values of all but the specified strata will be multiplied.
```

Comment:
you wanted to decrease opacity so this is confusing. Can you rephrase?

Reply:
I think this is the most accurate description of what it does. Yes, it allows only values between 0 and 1 as input and can therefore never increase the alpha, but technically it's just a multiplication.

See also the answer to your other comment:

> 
> I think the route of specifying the foreground stratas' alpha and then automatically reducing the others would be too strong of an opinion because the user might want to adjust this, depending on preference and colourscheme.
> 
> While I am decreasing the alpha of the background strata, I have written the code with add_CI in mind. Since the CI already have an alpha, I chose to go with a multiplier instead of subtraction. This way, both the line as well as the CI are made more transparent proportionally.

Reply:
I still don't agree:
- make argument name shorter
- make it more meaningful as to what is happening